lib/sasl.js

```diff
@@ -228,25 +228,39 @@ SaslClient.prototype.on_sasl_mechanisms = function (frame) {
         var mech = frame.performative.sasl_server_mechanisms[i];
         var f = this.mechanisms[mech];
         if (f) {
-            this.mechanism = f();
+            this.mechanism = typeof f === 'function' ? f() : f;
```

Comment:
Can you explain this a little? (Not saying it is wrong, just want to understand better).

Reply:
sure: in the current state of things, if I create my own custom SASL mechanism class, I have to let the client instantiate it with no parameters (it calls `f()` to instantiate it). In my case this design makes my life more difficult because I want to be able, when I instantiate the SASL mechanism, to give it a couple of parameters and a reference to the object that interfaces with the TPM chip.

I made this change so that if I give rhea an object that is already instantiated, it does not try to create it, it just takes it "as is". Does that make sense?

Not saying it's the right way to do it. this was the quick way :)

Reply:
makes sense, thanks!